### ISSUE TITLE
Widget magnifier not working/showing on product page for 2.4.5-p1

### ISSUE THREAD

**Issue by Rickertje** [magento/magento2#37194]
### Preconditions and environment

M2.4.5-p1

### Steps to reproduce

1) Clean install with sample data
2) Edit vendor/magento/theme-frontend-luma/etc/view.xml
3) Change var name enabled from false to true
4) Upgrade/clear cache etc
5) Visit the store and select a product fe: https://[magentoshopdomain]/index.php/breathe-easy-tank.html
6) Move the mouse over the product image.

### Expected result

Magnifier is shown

### Actual result

Magnifier is not shown

### Additional information
Also see: https://developer.adobe.com/commerce/frontend-core/javascript/jquery-widgets/magnifier/

_No response_

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [X] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
**Hi @Rickertje ,**

Verified the issue in 2.4-develop instance,But the issue is not reproducible.

**Preconditions:**
Magento Version 2.4-Develop
PHP Version 8.1

**Steps to reproduce:**
-  install with sample data
- Edit vendor/magento/theme-frontend-luma/etc/view.xml
- Change var name enabled from false to true
- Upgrade/clear cache.
- Visit the store and select a product front end
- Move the mouse over the product image.

**Kindly refer the below screenshots:**

<img width="1512" alt="var2" src="https://user-images.githubusercontent.com/60198592/224507270-34a2f4ce-b31f-46ca-908e-fafbdf374907.png">
<img width="1512" alt="var3" src="https://user-images.githubusercontent.com/60198592/224507272-a1e6749e-18b3-4d66-966f-97a6d90ab312.png">
<img width="1512" alt="var4" src="https://user-images.githubusercontent.com/60198592/224507287-25a2ef62-6c00-43cc-8db5-3950c4d843a2.png">
<img width="1512" alt="var1" src="https://user-images.githubusercontent.com/60198592/224507263-e2a1d1ab-d263-4d96-97a0-de771a7303ee.png">**

Magnifier is shown in the front-end and Customer can able perform Zoom in out Zoom out action.As per the description we are getting expected result.So the issue is not reproducible in 2.4-develop instance,Kindly provide more information if the issue is still reproducible.

Regards,

**Comment by Rickertje** [comment]
Sorry I was refering to the onpage magnifier that pops up when you hover over the product image on the product page (without clicking on the image). Can you check again? In your develop instance this is not working also.

![Knipsel1](https://user-images.githubusercontent.com/27420995/224508287-63ef61d2-a880-4f94-9ab0-303fb9e950c9.JPG)
![Knipsel2](https://user-images.githubusercontent.com/27420995/224508288-0bc929f2-d84d-4e07-b1b9-800ed4bed33b.JPG)

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
@mageto give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
Hi @Rickertje,

Thanks for your quick response,

We verified the issue in 2.4-develop branch as per your last [comments](https://github.com/magento/magento2/issues/37194#issuecomment-1464995314),Hence the issue is reproducible in 2.4-develop instance.Magnifier is not shown in Store front end when we select any particular product.But there is no Error appearing in Developer console.
Kindly refer the below screenshots:
<img width="1505" alt="mag1" src="https://user-images.githubusercontent.com/60198592/224894021-502ece42-351e-4a82-a3a3-ea7ef154269b.png">
<img width="1505" alt="MAG2" src="https://user-images.githubusercontent.com/60198592/224894154-367352e5-8a2d-4ce4-8dc9-f1c8d0b225dd.png">


**NOTE:** As per your last [comments](https://github.com/magento/magento2/issues/37194#issuecomment-1464995314)
the issue is reproducible in 2.3.7-p1 Magento instance,Now we are supporting magento version 2.4-develop,Kindly upgrade the magento version and check the issue again maybe you got some information from there.

Regards,

**Comment by Rickertje** [comment]
@magento give me 2.4.5-p1 instance

**Comment by Rickertje** [comment]
@magento give me 2.4.5-p1 instance

**Comment by Rickertje** [comment]
@magento give me 2.4.5-p1 instance

**Comment by Rickertje** [comment]
Hi @engcom-Dash. I'm unable to get a 2.4.5-p1 instance from @magento so I'm not 100% sure if the JMigrate warnings are present in 2.4.5-p1 (although they show up in my local clean install overhere). That is why I updated the problem description and removed my comment about these warnings. Furthermore:
I already updated my environment from 2.3.7-p1 to 2.4.5-p1 and that was when I dicovered that the magnifier widget is not working anymore. I see that you verified the problem on 2.4.5-p1. Do you have enough information to proceed? Thanks Rajasekar.

**Comment by engcom-Dash** [comment]
Hi @Rickertje ,

**_Issue confirmed !_**

Verified the issue in 2.4-develop instance and the issue is reproducible,Hence we are confirming the issue.

**_Preconditions:_**
Magento Version 2.4-develop
PHP version 8.1

**_Steps to reproduce:_**

- install with sample data
- Edit vendor/magento/theme-frontend-luma/etc/view.xml
- Change var name enabled from false to true
- Upgrade/clear cache.
- Visit the store and select a product front end
- Move the mouse over the product image.

**_Kindly refer the below screenshots:_**

![image](https://user-images.githubusercontent.com/60198592/225560269-2685d416-356b-4f60-a1a8-7901efc2ae20.png)
![image](https://user-images.githubusercontent.com/60198592/225560323-75b998c3-2435-44c2-b75b-5f8dad4333c9.png)

After changed var name enabled from false to true,The magnifier is not appearing product page,This is actual result as per the description,Hence we confirming the issue in 2.4-develop instance

Regards,

**Comment by Viper9x** [comment]
The reported issue affects Magento 2.4.5 and 2.4.6. It has already been fixed internally ACP2E-1648.
The fix is now available in the 2.4-develop branch https://github.com/magento/magento2/commit/9f62fc907385b6ddf546dae933519bd4da81e984

For anyone looking to apply the fix: we can quickly grab a patch from the git commit by adding ".patch" or ".diff" at the end of the URL: https://github.com/magento/magento2/commit/9f62fc907385b6ddf546dae933519bd4da81e984.patch

And here is the patch content:
```
From 9f62fc907385b6ddf546dae933519bd4da81e984 Mon Sep 17 00:00:00 2001
From: Alexandra Zota <[zota@adobe.com](mailto:zota@adobe.com)>
Date: Tue, 21 Feb 2023 11:19:37 +0200
Subject: [PATCH] ACP2E-1648: fix magnifier loading

---
 lib/web/magnifier/magnifier.js | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/lib/web/magnifier/magnifier.js b/lib/web/magnifier/magnifier.js
index 29726688f4d5f..8b525b2e047b8 100644
--- a/lib/web/magnifier/magnifier.js
+++ b/lib/web/magnifier/magnifier.js
@@ -541,11 +541,9 @@
 
             showWrapper = false;
             $(thumbObj).on('load', function () {
-                if (data.length > 0) {
+                if (data.hasOwnProperty(idx)) {
                     data[idx].status = 1;
-
                     $(largeObj).on('load', function () {
-
                         if (largeObj.width > largeWrapper.width() || largeObj.height > largeWrapper.height()) {
                             showWrapper = true;
                             bindEvents(eventType, thumb);
@@ -559,7 +557,6 @@
                             updateLensOnLoad(idx, thumb, largeObj, largeWrapper);
                         }
                     });
-
                     largeObj.src = data[idx].largeUrl;
                 }
             });

```
Document on how to apply a composer patch for Adobe Commerce (Magento) on Cloud infrastructure https://experienceleague.adobe.com/docs/commerce-knowledge-base/kb/how-to/how-to-apply-a-composer-patch-provided-by-magento.html?lang=en, and in more details here [https://experienceleague.adobe.com/docs/commerce-cloud-service/user-guide/develop/upgrade/apply-patches.html#:~:text=Copy-,Apply%20a%20custom%20patch,-When%20you%20deploy](https://experienceleague.adobe.com/docs/commerce-cloud-service/user-guide/develop/upgrade/apply-patches.html#:%7E:text=Copy-,Apply%20a%20custom%20patch,-When%20you%20deploy)

Document on how to apply a composer patch for Adobe Commerce (Magento) on a non-Cloud project: https://experienceleague.adobe.com/docs/commerce-operations/upgrade-guide/patches/apply.html

**Comment by engcom-Bravo** [comment]
Hi @Rickertje,

As per this comment https://github.com/magento/magento2/issues/37194#issuecomment-1554928574 issue has been fixed.Hence we are closing this issue.

Thanks.
